packages/utils/src/instrument/dom.ts

```diff
@@ -0,0 +1,260 @@
+/* eslint-disable @typescript-eslint/no-explicit-any */
+/* eslint-disable @typescript-eslint/ban-types */
+import type { HandlerDataDom } from '@sentry/types';
+
+import { uuid4 } from '../misc';
+import { addNonEnumerableProperty, fill } from '../object';
+import { GLOBAL_OBJ } from '../worldwide';
+import { addHandler, maybeInstrument, triggerHandlers } from './_handlers';
+
+type SentryWrappedTarget = HTMLElement & { _sentryId?: string };
+
+type AddEventListener = (
+  type: string,
+  listener: EventListenerOrEventListenerObject,
+  options?: boolean | AddEventListenerOptions,
+) => void;
+type RemoveEventListener = (
+  type: string,
+  listener: EventListenerOrEventListenerObject,
+  options?: boolean | EventListenerOptions,
+) => void;
+
+type InstrumentedElement = Element & {
+  __sentry_instrumentation_handlers__?: {
+    [key in 'click' | 'keypress']?: {
+      handler?: Function;
+      /** The number of custom listeners attached to this element */
+      refCount: number;
+    };
+  };
+};
+
+const WINDOW = GLOBAL_OBJ as unknown as Window;
+const DEBOUNCE_DURATION = 1000;
+
+let debounceTimerID: number | undefined;
+let lastCapturedEventType: string | undefined;
+let lastCapturedEventTargetId: string | undefined;
+
+/**
+ * Add an instrumentation handler for when a click or a keypress happens.
+ *
+ * Use at your own risk, this might break without changelog notice, only used internally.
+ * @hidden
+ */
+export function addClickKeypressInstrumentationHandler(handler: (data: HandlerDataDom) => void): void {
```

Comment:
this name is pretty explicit, alternatively we could do `addDomEventInstrumentationHandler`?